text/trace/0220-messaging-semantic-conventions-span-structure.md

```diff
@@ -0,0 +1,656 @@
+scenarios. In certain batching scenarios where many messages are created and
+published in large batches, creating a span for every single message would
+obfuscate traces and is not desirable. Thus instrumentation libraries and
+auto-instrumentation should default to creating a unique and distinct context
+per message, but may support configuration or other ways to change this default
+behavior. The latter can help to reduce the number of spans and to avoid overly
+verbose traces.
+
+For each producer scenario, a "Publish" span needs to be created. This span
+measures the duration of the call or operation that provides messages for
+sending or publishing to an intermediary. This call or operation (and the
+related "Publish" span) can either refer to a single message or to a batch of
+multiple messages.
+
+There are three different scenarios for injecting a creation context into a message:
+
+1. A user provides a custom creation context for a message, which doesn't refer
+   to any span described in this document. This provides flexibility to users
+   to model custom scenarios. In this case, no other spans besides the "Publish"
+   span should be created. The "Publish" span should link to the provided
+   creation context.
+2. If no custom creation context is provided for a message, it is recommended
+   to create a "Create" span for every single message and inject its context
+   into the message.  "Create" spans can be created during the "Publish" operation
+   as children of the "Publish" span.  Alternatively, "Create" spans can be
+   created independently of the "Publish" operation, e. g. in cases where messages
+   are created before they are passed to a "Publish" operation. In this case, the
+   "Publish" span should link to the "Create" spans.
+3. For single-message scenarios and when large number of spans are a problem,
+   the context of the "Publish" span can be injected into the message, thus
+   acting as the creation context. In this case, no other spans besides the
+   "Publish" span should be created.
+
+### Consumer
+
+For many use cases, it is not possible to rely on the presence of "Process"
+spans for correlating producer with consumer traces: there are cases where a
+dedicated processing operation cannot be identified, or where processing
+happens in a different trace. Furthermore, processing operations are not often
+covered by messaging libraries and SDKs, but take place in application code.
+Consistently creating spans for "Processing" operations would require either
+effort from the application owner to correctly instrument those operations, or
+additional capabilities of messaging libraries and SDKs (e. g. hooks for
+processing callbacks, which can then be instrumented by the libraries or SDKs).
+
+While it is possible to create "Process" spans and correlate those with
+consumer traces in certain cases, this is not something that can be generally
```

Comment:
Is "producer" meant here? (Cf. previous, maybe "or where processing should happen in the _same_ trace was meant?)